app/tools/skills/alloy_discovery.py

```
@@ -0,0 +1,330 @@
+        try:
+            from app.tools.calphad import _analyze_phases  # type: ignore
+            for row in rows[: min(3, len(rows))]:  # top 3 only — CALPHAD is slow
+                if "formula" not in row:
+                    continue
+                try:
+                    row["phases"] = _analyze_phases(
+                        composition=row["formula"],
+                        temperature=service_temperature_K,
+                        database="TCHEA",
+                    )
+                except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)